#164552962 Hosted the Backend on Heroku and Implemented LFA Feedback

ola357 opened the pull request:
#### What does this PR do?
Implement LFA(Product Owner) Feedback
Host Server(Backend) on Heroku
#### Description of Task to be completed?
Implement ES6 Classes
Object destructuring of req.body in POST requests
Create Remote app on Heroku
Configure package.json
#### How should this be manually tested?
GET https://murmuring-cove-64845.herokuapp.com/api/v1/messages
GET https://murmuring-cove-64845.herokuapp.com/api/v1/messages/unread
GET https://murmuring-cove-64845.herokuapp.com/api/v1/messages/sent
GET https://murmuring-cove-64845.herokuapp.com/api/v1/messages/<:message-id>
POST https://murmuring-cove-64845.herokuapp.com/api/v1/messages/
POST https://murmuring-cove-64845.herokuapp.com/api/v1/auth/signup
POST https://murmuring-cove-64845.herokuapp.com/api/v1/auth/login
DELETE https://murmuring-cove-64845.herokuapp.com/api/v1/messages/<:message-id>
GET messages

#### What are the relevant pivotal tracker stories?
#164552962

ola357 committed:
feat(developer): I installed eslint
-extends the airbnb style guide
[Delivers #164486707]

ola357 committed:
feat(developer): setup babel
-to transpile from es6 to es5
[Delivers #164486291]

ola357 committed:
feat(developer): set up mocha
-for unit testing
[Delivers #164486992]

ola357 committed:
feat(user): setup the server using express framework
[Delivers #164487943]

ola357 committed:
feat(developer): the get all recieved messages
-currently failing its unit test
-following the Test Driven Development approach
[Delivers #164516793]

ola357 committed:
feat(user): get message route
-fetches all recieved messages for the user
[Delivers #164487450]

ola357 committed:
feat(developer): integrated travis
-for continous integration and
-integrated tests
[Delivers #164510155]

ola357 committed:
feat(developer): itegrated coveralls
-for test code coverage reporting
[Delivers #164516851]

ola357 committed:
feat(user): The user signup endpoint is built
-7 other endpoints are built
-tested with npm test and tests passing
-tested the api endpoints with postman
[Delivers #164487368]

ola357 committed:
feat(developer): restructured folders to follow MVC
-added CI badges from Travis and Coveralls to README
[Delivers #164537581]

ola357 committed:
feat(developer): Implemented LFA feedback
-Used ES6 classes and static methods
-Used MVC structure
-Destructured req.body in my POST routes
-prepared the app for hosting on heroku
[Delivers #164552962]

ola357 committed:
fixing error

ola357 committed:
fixing error 2
-changed the start script
-after npm run build & npm run serve

ola357 committed:
fixing error 3
-required @babel/polyfill
-in authControllers.js

ola357 committed:
fixing error 4
-added require @babel/register

ola357 committed:
fixing error 5
removed the required @babel polyfill & register
-added require @babelregister to app.js

ola357 committed:
fixing error 6
-added @babelcore

ola357 committed:
feat(developer): changed imported babelpolyfill
-in app.js

cwizard2011 commented on server/controllers/authControllers.js:
```diff
@@ -0,0 +1,50 @@
+
+const engine = encrypter('my secret');
+
+class authControllers {
```
Class should always start with Capital letter

cwizard2011 replied:
Same with the file name. authControllers.js should be AuthController.js

cwizard2011 commented on server/controllers/messagesController.js:
```diff
@@ -0,0 +1,88 @@
+import validate from '../validators/validate';
+
+
+class messagesController {
```
class name should start with capital

cwizard2011 commented on server/validators/validate.js:
```diff
@@ -0,0 +1,32 @@
+import Joi from 'joi';
+
+
+class validate {
```
Class starts with capital letter

cwizard2011 commented on test/auth.js:
```diff
@@ -0,0 +1,110 @@
+        .post('/api/v1/auth/signup')
+        .send(user)
+        .end((err, res) => {
+          res.should.have.status(200);
```
You can always test for more things in your test, not only the status

ola357 committed:
feat(developer): Implemented LFA feedback
_made classes Start with Capital Letter
-added more tests

ola357 committed:
fix(developer): Trying to fix error I get on Travis
-test are succesfully locally
-but they fail in travis
-just added clean to my scripts
